java-checks/src/main/java/org/sonar/java/checks/MethodParametersOrderCheck.java

```diff
@@ -98,28 +99,32 @@ public static boolean allUnique(List<String> argListNames) {
     return argListNames.size() == new HashSet<>(argListNames).size();
   }
 
-  private static boolean matchingTypesWrongOrder(ParametersList formalParameterList, List<IdentifierTree> argumentList) {
+  private static List<VariableTree> matchingTypesWrongOrder(ParametersList formalParameterList, List<IdentifierTree> argumentList) {
     Iterator<IdentifierTree> argumentsIterator = argumentList.stream().filter(Objects::nonNull).iterator();
-    int countArgumentsNotOrdered = 0;
+    List<VariableTree> misplacedParameters = new ArrayList<>();
     while (argumentsIterator.hasNext()) {
       IdentifierTree argument = argumentsIterator.next();
       int index = formalParameterList.indexOf(argument.name().toLowerCase(Locale.ENGLISH));
       Type formalType = formalParameterList.typeOfIndex(index);
       Type argType = argument.symbolType();
       if (!formalType.is(argType.fullyQualifiedName()) || formalType.isUnknown() || argType.isUnknown()) {
-        return false;
+        return new ArrayList<>();
```

Comment:
I would prefer to use `Collections.emptyList()` here to have the returned collection unmodifiable by other methods.